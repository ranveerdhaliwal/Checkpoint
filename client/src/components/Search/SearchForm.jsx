import React from 'react';
import PropTypes from 'prop-types';

import { TextField } from 'components/common';

import IconButton from '@material-ui/core/IconButton';
import SearchIcon from '@material-ui/icons/Search';

import styled from 'styled-components';

const FormWrapper = styled.div`
  display: flex;
  align-items: flex-end;
`;

const TextFieldWrapper = styled.div`  
  width: 100%;
`

const setTextState = (newText) => ({
  searchText: newText
});

class SearchForm extends React.Component {
  state = {
    searchText: ''
  };

  handleTextChange = (event) => {
    this.setState(setTextState(event.target.value));
  }

  handleSearch = () => {
    this.props.searchForGame(this.state.searchText);
  }

  handleKeyPress = (event) => {
    if (event.key === 'Enter') {
      this.handleSearch();
    }
  }

  render() {
    return (
      <FormWrapper>
        <TextFieldWrapper>
          <TextField
            label="Search for games"
            onChange={this.handleTextChange}
            onKeyPress={this.handleKeyPress}
            value={this.state.searchText}
            >
          </TextField>
        </TextFieldWrapper>
        <IconButton onClick={this.handleSearch} color="primary" aria-label="Search">
          <SearchIcon />
        </IconButton>
      </FormWrapper>
    )
  }
}

SearchForm.propTypes = {
  searchForGame: PropTypes.func.isRequired,
}

export default SearchForm;