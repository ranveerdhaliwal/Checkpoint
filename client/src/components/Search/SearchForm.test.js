import React from 'react';
import { shallow } from 'enzyme';

import SearchForm from './SearchForm';

describe('Search Form', () => {
  let component;
  let instance;

  const props = {
    searchForGame: jest.fn()
  };

  beforeEach(() => {
    component = shallow(<SearchForm {...props} />);
    instance = component.instance();
  });

  it('snapshot test', () => {
    expect(component).toMatchSnapshot();
  });


  it('updates search term state properly', () => {
    expect(instance.state.searchText).toEqual('');

    const newSearchTerm = 'mario bros';
    instance.handleTextChange(
      {target: 
         {value: newSearchTerm}
       }
    );
    expect(instance.state.searchText).toEqual(newSearchTerm);

  });

  it('calls dispatch function ', () => {
    const newSearchTerm = 'halo';
    instance.handleTextChange(
      {target: 
         {value: newSearchTerm}
       }
    );
    expect(instance.state.searchText).toEqual(newSearchTerm);

    const button = component.find('WithStyles(IconButton)');
    button.props().onClick();

    expect(props.searchForGame).toHaveBeenCalled();

    expect(props.searchForGame).toBeCalledWith('halo');
  });

});
