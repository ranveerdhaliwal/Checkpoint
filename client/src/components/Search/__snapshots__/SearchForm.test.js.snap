// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form snapshot test 1`] = `
<styled.div>
  <styled.div>
    <TextField
      autoFocus={true}
      label="Search for games"
      onChange={[Function]}
      onKeyPress={[Function]}
      value=""
    />
  </styled.div>
  <WithStyles(IconButton)
    aria-label="Search"
    color="primary"
    onClick={[Function]}
  >
    <pure(SearchIcon) />
  </WithStyles(IconButton)>
</styled.div>
`;
